extensions/Worker.Extensions.Shared/Exceptions/InvalidContentTypeException.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions
+{
+    /// <summary>
+    /// The exception that is thrown when an invalid content-type is provided.
+    /// </summary>
+    internal class InvalidContentTypeException : InvalidOperationException
+    {
+        /// <summary>
+        /// Initializes a new instance of the InvalidContentTypeException class with a specified error message.
+        /// </summary>
+        /// <param name="contentType">The content type(s) that is supported.</param>
+        public InvalidContentTypeException(string contentType) : base($"Unexpected content-type. Only '{contentType}' is supported.") { }
```

Comment:
Same as with the source comment above, it would be helpful to pass the "actual" content type used.